vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/synchronization/index.js

```diff
@@ -62,7 +75,31 @@ function synchronizeFeature(target, source, context) {
   }
 
   target.setGeometry(context.lookup.get(source.geometry));
-  target.setStyle(context.lookup.get(source.style));
+
+  // Define style function that is run before rendering each feature. The
+  // function supports using a fallback text style for rendering labels in case
+  // the feature doesn't define its own text style.
+  // Acquire reference to style instance outside of style function, otherwise
+  // there would be no reference to the instance, and it might get garbage
+  // collected.
+  const style = context.lookup.get(source.style);
+  target.setStyle(() => {
+    if (!style) {
+      return undefined;
+    }
+    // If feature has a label but no custom text style, then use default text
+    // style
+    if (source.label && !style.getText()) {
+      style.setText(fallbackTextStyle);
+    }
+    // Set the feature's label on the text style. This is safe even when using
+    // the default text style instance, as for each feature using the default
+    // text style, this function will be called again before rendering.
+    if (style.getText()) {
+      style.getText().setText(source.label);
+    }
+    return style;
+  });
```

Comment:
This is more involved than it needs to be at the moment, but it is a starting point for adding customization of the label's styles.